lib/dao.js

```diff
@@ -2638,8 +2638,12 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
   assert(typeof cb === 'function', 'The cb argument must be a function');
 
   var connector = this.getConnector();
-  assert(typeof connector.replaceById === 'function',
-          'replaceById() must be implemented by the connector');
+
+  if (typeof connector.replaceById !== 'function') {
+    var err =  new Error('there is no support for replaceById() in the connector.' +
+      'Connector needs to add support for `replaceById()`');
+    return cb(err);
+  }
```

Comment:
@Amir-61 Spoke to @bajtos over slack, your change LGTM.

TLDR: We should make all the other asserts operate in the same fashion as yours, but it's prob not worth the effort to go change all the other ones. We can discuss more during actual DAO refactorings.